pkg/ipam/strictvl3ipam/server.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2024 Cisco and its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package strictvl3ipam provides a networkservice.NetworkService Server chain element that resets IP context configuration out of the settings scope
+package strictvl3ipam
+
+import (
+	"context"
+	"net"
+	"sync"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/ipam"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/ippool"
+)
+
+type strictVl3IPAMServer struct {
+	ipPool *ippool.IPPool
+	m      sync.Mutex
+}
+
+// NewServer - returns a new ipam networkservice.NetworkServiceServer that validates the incoming IP context parameters and resets them based on the validation result.
+func NewServer(ctx context.Context, prefixCh <-chan *ipam.PrefixResponse) networkservice.NetworkServiceServer {
+	var ipPool = ippool.New(net.IPv4len)
+
+	s := &strictVl3IPAMServer{ipPool: ipPool}
+	go func() {
+		for prefix := range prefixCh {
+			s.m.Lock()
+			s.ipPool = ippool.NewWithNetString(prefix.Prefix)
+			s.m.Unlock()
+		}
+	}()
+
+	return s
+}
+
+func (n *strictVl3IPAMServer) areAddressesValid(addresses []string) bool {
+	n.m.Lock()
+	defer n.m.Unlock()
+
+	for _, addr := range addresses {
+		if !n.ipPool.ContainsNetString(addr) {
```

Comment:
What if we have multiple addresses? For example, IPv4 and IPv6? Or multiple IPv4?
I would suggest that addresses are valid if at least one is valid.